draft-ietf-quic-transport.md

```diff
@@ -2427,6 +2407,32 @@ ACK Block (repeated):
   packets preceding the largest packet number, as determined by the
   preceding Gap.
 
+### Sending ACK Frames
+
+Implementations MUST NOT generate packets that only contain ACK frames in
+response to packets which only contain ACK frames. However, they SHOULD
+acknowledge packets containing only ACK frames when sending ACK frames in
+response to other packets.  Implementations MUST NOT send more than one ACK
+frame per received packet that contains frames other than ACK frames.  Packets
```

Comment:
This doesn't make sense to me:

I can receive a packet that contains other than ACK frames (which is the norm, I would expect).  But I might send more than one ACK frame over time.  I think that what you intend to say here is part of a different statement.

That is, every time a packet that contains more than just ACK frames is received, an implementation schedules an ACK frame to be sent in a new packet.  This might be immediately, but more often it will be delayed slightly.  Delaying acknowledgment in this way ensures that packets arriving in rapid succession only induce a single packet in response or that acknowledgments can be sent together with stream data.  Either way, fewer packets are sent, and fewer packets containing only ACK frames.  Packets containing only ACK frames that are generated as a result of incoming packets MUST NOT be generated at a rate higher than the rate of incoming packets.  Otherwise there would be in amplification in the number of packets, if not the number of bytes being exchanged.

Reply:
This is new text, but I think it's correct.  There's no case where receiving one packet generates two ack frames.